Islandora/Crayfish
# Versioning

**GitHub Issue**: (https://github.com/Islandora-CLAW/CLAW/issues/740)

* https://github.com/Islandora-CLAW/Alpaca/pull/61 and https://github.com/Islandora-CLAW/islandora/pull/154

# What does this Pull Request do?
Fires off a call to the chullo Fedora API to create a version when the version event type is received from Alpaca

# What's new?
This method reads the event type header sent in from Alpaca on the POST node stream to the saveNode method. It passes that header value to the service's saveNode method which then looks at the value. If the event type is "Version" then it invokes chullo's fedoraApi methods for creating a version.
Todo: I need to write tests. I will do so once I'm sure this is the right approach to take.

# How should this be tested?
* update Milliner with this PR
* probably want to have the Alpaca PR and Islandora PR mentioned above in place, otherwise you could hard code the value of $event_type = "Version" in the saveNode method (but that means you won't get node updates in fedora just so you know...)
* make sure you get a version created in Fedora (note that when i did this, i got several versions within a few milliseconds of each other which i'll assume has to do with how many times the node update hook is being fired in Drupal...)

# Additional Notes:
As I mentioned in the PR for Alpaca, I could easily see this being an entire separate end point in Milliner which a separate route and a separate method from the create/update POST endpoint. If that is what would be considered, I can make modifications for that. The way it's written right now a request to milliner's POST node/uuid route does 1 of 3 things: create node, create version of node, or update node. It might be slightly less tangled if it was its own endpoint that didn't have to read from a header.

# Interested parties
@Islandora-CLAW/committers
